modules/js-and-css/optimize-oembeds/hooks.php

```diff
@@ -0,0 +1,99 @@
+<?php
+/**
+ * Hook callbacks used for oEmbed Optimizer.
+ *
+ * @since n.e.x.t
+ * @package performance-lab
+ */
+
+if ( ! defined( 'ABSPATH' ) ) {
+	exit; // Exit if accessed directly.
+}
+
+$oembed_lazy_load_scripts = false;
+
+/**
+ * Filter the oEmbed HTML.
+ *
+ * Add loading="lazy" to any iframe tags.
+ * Lazy load any script tags.
+ *
+ * @since n.e.x.t
+ *
+ * @param string $html The oEmbed HTML.
+ * @return string
+ */
+function plab_optimize_oembed_html( $html ) {
+	global $oembed_lazy_load_scripts;
+
+	// Find script tags in the oembed using regex.
+	preg_match_all( '/<script\b[^>]*>([\s\S]*?)<\/script>/mi', $html, $matches );
```

Comment:
Any chance we could consider using the HTML API instead of regexes, for this module? It feels like a missed opportunity to introduce new regexes, when [we know all the issues they come with](https://make.wordpress.org/core/2023/08/19/progress-report-html-api/#why-the-html-api).

Since we don't appear to be targeting older WP versions, this seems like the perfect opportunity to do it, as oEmbeds aren't as common as other hooks for which we've considered using the API (such as image processing). This means we won't be introducing unreasonable processing delays — even though the HTML API has pretty good performance at this point, to my knowledge.

CC @dmsnell

Reply:
switched to WP_HTML_Tag_Processor in https://github.com/WordPress/performance/pull/886/commits/21f29082121061704e84d58a185b79d60b7198c8

Reply:
Created a Gist with the regex version which could be useful for optimization plugins that want to ensure compatibility with older WordPress versions - https://gist.github.com/adamsilverstein/413bc328b3ae01113085fb9edabce963